api/v1alpha1/conversion.go

```diff
@@ -32,7 +32,21 @@ func (src *Rollout) ConvertTo(dst conversion.Hub) error {
 	switch t := dst.(type) {
 	case *v1beta1.Rollout:
 		obj := dst.(*v1beta1.Rollout)
+
+		// metadata
 		obj.ObjectMeta = src.ObjectMeta
+		if src.Spec.DeprecatedRolloutID != "" {
+			if obj.Labels == nil {
+				obj.Labels = map[string]string{}
+			}
+			obj.Labels[v1beta1.RolloutIDLabel] = src.Spec.DeprecatedRolloutID
+		}
+		if obj.Annotations == nil {
+			obj.Annotations = map[string]string{}
+		}
+		obj.Annotations[RolloutStyleAnnotation] = src.Annotations[RolloutStyleAnnotation]
```

Comment:
why copy the annotation to v1beta1 , v1beta1 should save the information in the `Spec.Strategy.Canary.EnableExtraWorkloadForCanary`